src/fpm/dependency.f90

```diff
@@ -349,28 +363,49 @@ subroutine add_dependencies(self, dependency, error)
     end if
 
     do ii = 1, size(dependency)
-      call self%add(dependency(ii), error)
+      call self%add(dependency(ii), error, parent=parent)
       if (allocated(error)) exit
     end do
     if (allocated(error)) return
 
   end subroutine add_dependencies
 
   !> Add a single dependency to the dependency tree
-  pure subroutine add_dependency(self, dependency, error)
+!  pure subroutine add_dependency(self, dependency, error, parent)
+  subroutine add_dependency(self, dependency, error, parent)
     !> Instance of the dependency tree
     class(dependency_tree_t), intent(inout) :: self
     !> Dependency configuration to add
     type(dependency_config_t), intent(in) :: dependency
     !> Error handling
     type(error_t), allocatable, intent(out) :: error
+    !> Name of parent package
+    character(len=*), intent(in), optional :: parent
 
-    integer :: id
+    integer :: id, i, parent_id
+    logical :: found
 
+    parent_id = 0
+    found = .false.
+    if (present(parent)) then
+      parent_id = self%find(parent)
+      if (parent_id < 1) call fpm_stop(1,'*add_dependency*:Error: No such package in dependency tree.')
```

Comment:
Better create an error here and propagate up. Stop should only be called by the top-level command, if the error cannot be recovered.